tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -970,6 +971,10 @@ create_compressed_scan_paths(PlannerInfo *root, RelOptInfo *compressed_rel, int
 {
 	Path *compressed_path;
 
+	/* clamp total_table_pages to 10 pages since this is the
+	 * minimum estimate for number of pages */
+	root->total_table_pages = Max(compressed_rel->pages, 10);
```

Comment:
Maybe use some PostgreSQL macro or variable for that instead of hard coding the value?